### PR TITLE
Check for proper whitespaces for function return types

### PR DESCRIPTION
**Pull request by hambai**
We noticed that there is WPCS rule for spacing around the colon in function return types. We had some discrepancies in how we were spacing the colon and since most of our team is using the Neutron coding standards I decided we should implement that check too.

The idea is that there should be no space between the function declaration and the colon, but then there should be a space between the colon and the return type.

So basically:
```
// valid
function(): bool {...
// invalid
function() : bool {...
// invalid
function() :bool {...
```

see the "return type" example here - https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#space-usage

### CODE REVIEW

File: tests/Sniffs/Functions/FunctionsFixture.php [hunk 1]
```diff
@@ -135,7 +135,7 @@ public function missingArgHintThree(string $arg1, $arg2): string {
 	}
 
 	// Next line should warn about unused type hint
-	public function hasReturnHintButNoReturn() : string {
+	public function hasReturnHintButNoReturn(): string {
```
Comment (sirbrillig) [inline tests/Sniffs/Functions/FunctionsFixture.php:138]: Maybe we should add a test to this file that uses the colon incorrectly as well, since the Closures tests only test anonymous functions.

File: tests/Sniffs/Functions/ClosureFixture.php [hunk 2]
```diff
@@ -16,8 +16,29 @@ public function hasNoReturnHintAndClosure() {
 	}
 
 	public function hasClosureWithReturn() {
+		$myFunc = function (): bool {
+			return true;
+		};
+	}
+
+	// Next line should warn about type hint spacing
```
Comment (sirbrillig) [inline tests/Sniffs/Functions/ClosureFixture.php]: The error will appear two lines down, not on the next line.

File: tests/Sniffs/Functions/ClosureFixture.php [hunk 3]
```diff
@@ -16,8 +16,29 @@ public function hasNoReturnHintAndClosure() {
...
+	public function hasClosureWithReturnAndWrongSpacing() {
 		$myFunc = function () : bool {
 			return true;
 		};
 	}
+
+	// Next line should warn about type hint spacing
```
Comment (sirbrillig) [inline tests/Sniffs/Functions/ClosureFixture.php]: The error will appear two lines down, not on the next line.

**Review by sirbrillig — COMMENTED** [review]
Looks good to me. I made a suggestion to add a few more tests, but otherwise 👍

File: tests/Sniffs/Functions/FunctionsFixture.php (hunk 1 above)
Comment (hambai) [reply tests/Sniffs/Functions/FunctionsFixture.php:138]: I've added more tests in the FunctionsFixture.php file

**Comment by hambai** [comment]
Since we're at it I decided to add a check that there is a space after the return type

**Comment by sirbrillig** [comment]
Nice work.
